File: su/2026-04/22_boj_문자열폭발.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

class Main {
// public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String s = br.readLine();
        String bomb = br.readLine();

        int n = s.length();
        int m = bomb.length();

        char[] stack = new char[n];
        int top = 0;

        char last = bomb.charAt(m - 1);

        for (int i = 0; i < n; i++) {
            char cur = s.charAt(i);
            stack[top++] = cur;

            // 마지막 문자가 같고, 길이가 충분할 때만 검사
            if (cur == last && top >= m) {
                boolean isBomb = true;

                for (int j = 0; j < m; j++) {
                    if (stack[top - m + j] != bomb.charAt(j)) {
                        isBomb = false;
                        break;
                    }
                }

                if (isBomb) {
                    top -= m; // 폭발 문자열 길이만큼 제거
                }
            }
        }

        if (top == 0) {
            System.out.println("FRULA");
        } else {
            System.out.println(new String(stack, 0, top));
        }
    }
}

File: su/2026-04/22_boj_문자열폭발.py
s = input().strip()
bomb = input().strip()
stack = []
m = len(bomb)
# def explode():
#     return

for ch in s:
    stack.append(ch)
    if stack[-1] == bomb[-1] and len(stack) >= m:
        if ''.join(stack[-m:]) == bomb:  
            del stack[-m:]  # stack 뒤에서 m개 원소 한번에 삭제

if stack:
    print(''.join(stack))
else:
    print('FRULA')

'''
del lst[:2]    # 앞에서 2개 삭제
del lst[-2:]   # 뒤에서 2개 삭제
'''